File: apps/client/src/types/radarr.ts
export interface RadarrSettings {
    id: number;
    api_key?: string;
    base_url?: string;
    minimum_availability?: string;
    quality?: number;
    root_folder?: number;
}

export interface RadarrRootFolder {
    id: number;
    path: string;
    accessible: boolean;
    free_space: number;
    unmapped_folders: RadarrRootFolderUnmappedFolder[];
}

export interface RadarrRootFolderUnmappedFolder {
    name: string;
    path: string;
    relative_path: string;
}

export interface RadarrQualityProfile {
    id: number;
    name: string;
    upgradeAllowed: boolean;
    cutoff: number;
    minFormatScore: number;
    cutoffFormatScore: number;
    items: RadarrQualityProfileItem[];
    language: RadarrQualityProfileLanguage;
}

export interface RadarrQualityProfileItem {
    quality: RadarrQuality;
    allowed: boolean;
}

export interface RadarrQuality {
    id: number;
    name: string;
    source: string;
    resolution: number;
    modifier: string;
}

export interface RadarrQualityProfileLanguage {
    id: number;
    name: string;
}

File: apps/server/pkg/structures/radarr.go
package structures

type RadarrSettings struct {
	ID                  int     `json:"id"`
	APIKey              *string `json:"api_key"`
	URL                 *string `json:"base_url"`
	MinimumAvailability *string `json:"minimum_availability"`
	Quality             *int    `json:"quality"`
	RootFolder          *int    `json:"root_folder"`
}

type RadarrQualityProfile struct {
	ID                int                          `json:"id"`
	Name              string                       `json:"name"`
	UpgradeAllowed    bool                         `json:"upgrade_allowed"`
	Cutoff            int                          `json:"cutoff"`
	MinFormatScore    int                          `json:"min_format_score"`
	CutoffFormatScore int                          `json:"cutoff_format_score"`
	Items             RadarrQualityProfileItem     `json:"items"`
	Language          RadarrQualityProfileLanguage `json:"language"`
}

type RadarrQualityProfileItem []RadarrQualityProfileItemData

type RadarrQualityProfileItemData struct {
	Quality RadarrQuality `json:"quality"`
	Allowed bool          `json:"allowed"`
}

type RadarrQuality struct {
	ID         int    `json:"id"`
	Name       string `json:"name"`
	Source     string `json:"source"`
	Resolution int    `json:"resolution"`
	Modifier   string `json:"modifier"`
}

type RadarrQualityProfileLanguage struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
}

type RadarrRootFolder struct {
	ID              int                              `json:"id"`
	Path            string                           `json:"path"`
	Accessible      bool                             `json:"accessible"`
	FreeSpace       int64                            `json:"free_space"`
	UnmappedFolders []RadarrRootFolderUnmappedFolder `json:"unmapped_folders"`
}

type RadarrRootFolderUnmappedFolder struct {
	Name         string `json:"name"`
	Path         string `json:"path"`
	RelativePath string `json:"relative_path"`
}
